lib/Guides/Compiler/SidebarTransformer.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Website\Guides\Compiler;
+
+use Override;
+use phpDocumentor\Guides\Compiler\CompilerContext;
+use phpDocumentor\Guides\Compiler\NodeTransformer;
+use phpDocumentor\Guides\Nodes\Menu\TocNode;
+use phpDocumentor\Guides\Nodes\Node;
+use phpDocumentor\Guides\RestructuredText\Nodes\GeneralDirectiveNode;
+
+/** @implements NodeTransformer<TocNode|GeneralDirectiveNode> */
+final class SidebarTransformer implements NodeTransformer
```

Comment:
What's the purpose of this class? It would be great to have some comments on classes that don't have an obvious purpose such as this one.